design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane ecosystem.
+- Allow users to fetch data from existing resources without giving full ownership.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
+
+## Proposal
+
+To support observing resources without taking ownership, we will introduce a new spec named `managementPolicy` to the
+Managed Resources. We will also deprecate the existing `deletionPolicy` in favor of the new spec since they will be
+controlling the same behavior; that is, how should the changes on the CR affect the external cloud resource.
+
+This new policy will have the following values:
+
+- `FullControl`(Default): Crossplane will fully manage and control the external resource, including deletion when the CR
+is deleted (same as `deletionPolicy: Delete`).
+- `ObserveCreateUpdate`: Crossplane will observe and perform create and update operations on the external resource but
+will not make any deletions (same as `deletionPolicy: Orphan`).
+- `ObserveOnly`: Crossplane will only observe the external resource and will not make any changes or deletions.
+
+Please note while `ObserveCreateUpdate` may sound verbose, it accurately describes the actions that Crossplane will take
+on the external resource. This naming convention also focuses on what Crossplane will do, rather than what it won't do,
+making it more intuitive and extensible for future policy options, such as `ObserveDelete`.
+
+As indicated above, `FullControl` and `ObserveCreateUpdate` policies will behave precisely the same as the deletion
+policies we have today, including keeping the default behavior the same. We will introduce the new behavior with
+the `ObserveOnly` option, which would be pretty similar to what we have today to [import existing managed resources],
+but instead of starting to manage after import, we will not make any modifications to the external resource and only
+sync status back.
+
+```yaml
+apiVersion: ec2.aws.crossplane.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    crossplane.io/external-name: vpc-12345678
+  name: observe-vpc
+spec:
+  managementPolicy: ObserveOnly
+  forProvider:
+    region: us-east-1
+```
+
+### Implementation
+
+Crossplane providers already manage external resources by implementing the Crossplane runtime's ExternalClient interface,
+which includes the four methods listed below.
+
+```go
+type ExternalClient interface {
+	Observe(ctx context.Context, mg resource.Managed) (ExternalObservation, error)
+	Create(ctx context.Context, mg resource.Managed) (ExternalCreation, error)
+	Update(ctx context.Context, mg resource.Managed) (ExternalUpdate, error)
+	Delete(ctx context.Context, mg resource.Managed) error
+}
+```
+
+We will leverage the fact that we have an already implemented Observe method for all managed resources by calling only
```

Comment:
Would we also need updates to the `Observe` implementations to handle observing fields that have historically been required?

Reply:
I think we will also need to update late-init logic to include required fields as well which worth mentioning 👍 

FWIW, this could be done once instead of per resource with code generation solutions like upjet.